tools/toolchain/dbuild

```diff
@@ -154,10 +154,8 @@ if [ -z "$is_podman" ]; then
        --pids-limit -1
        )
 else
-    TMP_PASSWD=$(mktemp --tmpdir passwd.XXXXXX)
-    FULLNAME=$(getent passwd $USER | cut -d ':' -f 5)
-    echo "$USER:x:0:0:$FULLNAME:$HOME:/bin/bash" > "$TMP_PASSWD"
-    docker_common_args+=(-v "$TMP_PASSWD:/etc/passwd:ro")
+    docker_common_args+=(
+        --userns keep-id)
```

Comment:
This breaks sudo, doesn't it? It's common to want to install stuff inside the container, but if you're not root, you can't.

Perhaps --userns keep-id:uid=0,gid=0 should work, but I get an error when I try it.

Reply:
It doesn't break sudo, because the fedora image has a catch-all permission in `/etc/sudoers`.
(`ALL ALL=(ALL:ALL) NOPASSWD: ALL`).

Reply:
If you want to map the external user to root, you have to use `--userns host` instead of `--userns keep=id:uid=0,gid=0`. IIUC the latter conflicts with some of podman's implementation details about intermediate mappings or something.

Reply:
@avikivity hi Avi, i concur with Michał. let me explain this wit a session in the dbuild. i started using the `open-coredump.sh`:
```console
[kefu@fedora ~]$ sudo grep 'ALL ALL' /etc/sudoers
ALL ALL=(ALL:ALL) NOPASSWD: ALL
[kefu@fedora ~]$ sudo dnf install -y -q neovim

Installed:
  compat-lua-libs-5.1.5-18.fc34.x86_64 libtermkey-0.20-10.fc34.x86_64 libtree-sitter-0.20.0-2.fc34.x86_64 libvterm-0.1.4-1.fc34.x86_64 luajit-2.1.0-0.19beta3.fc34.x86_64 luajit2.1-luv-1.41.1.0-2.fc34.x86_64 msgpack-3.1.0-7.fc34.x86_64
  neovim-0.5.1-1.fc34.x86_64           unibilium-2.0.0-8.fc34.x86_64  xsel-1.2.0-28.fc34.x86_64          
$ cat /etc/redhat-release 
Fedora release 34 (Thirty Four)
```

in the host:
```console
$ podman ps
CONTAINER ID  IMAGE                                                              COMMAND     CREATED        STATUS        PORTS       NAMES
dba726567965  docker.io/scylladb/scylla-toolchain:fedora-34-enterprise-20210905  bash -l     8 minutes ago  Up 8 minutes              awesome_franklin
```

please note, i am debugging an ancient branch, so the image is based on f34. but the `dbuild` script is from master HEAD with this patch applied.

Reply:
@avikivity did you get a chance to read the comment above? does this help?

Reply:
Ah, that was actually done for docker, which does keep the same uid.